### PR TITLE
Core 811 fix select all

### PR DESCRIPTION
**Pull request by roger-in-kiva**

### CODE REVIEW

File: src/components/LoansByCategory/QuickFilters/CheckboxList.vue [hunk 1]
```diff
@@ -78,7 +78,11 @@ export default {
 					if (exists) this.selected.splice(index, 1);
 				} else if (!exists) this.selected.push(item.value);
 			});
-			this.updateSelected(this.selected, undefined, true);
+			if (isAll && typeof item === 'object') {
```
Comment (mcstover) [inline src/components/LoansByCategory/QuickFilters/CheckboxList.vue:81]: `item` wouldn't be defined here at all right?

Comment (roger-in-kiva) [reply src/components/LoansByCategory/QuickFilters/CheckboxList.vue]: Logic updated, thanks!

File: src/components/LoansByCategory/QuickFilters/CheckboxList.vue [hunk 2]
```diff
@@ -75,13 +75,16 @@ export default {
 				const index = this.selected.indexOf(item.value);
 				const exists = index !== -1;
 				if (isAll) {
-					if (exists) this.selected.splice(index, 1);
-				} else if (!exists) this.selected.push(item.value);
+					if (exists) {
+						this.updateSelected(this.selected, item.value);
```
Comment (dyersituations) [inline src/components/LoansByCategory/QuickFilters/CheckboxList.vue:79]: It looks like the component is emitting an event for each item that was selected when select all is toggled. There should only be one event emitted for a user action.

Comment (roger-in-kiva) [reply src/components/LoansByCategory/QuickFilters/CheckboxList.vue:79]: Yeah, this is b/c the updater on parent was planned for one event. Selecting all is the less probable action here, main idea is to select a country you are interested.
We need to launch this asap, but if you believe a refactor is needed for this experiment I can fix it.

Comment (dyersituations) [reply src/components/LoansByCategory/QuickFilters/CheckboxList.vue:79]: Gotcha. If it works alright in functional testing, please add a `TODO` here to refactor into one emitted event. In general you really only want one distinct event per user action, unless there's a chain of actions. Here we have one user interaction with one result (reset filter).

Comment (roger-in-kiva) [reply src/components/LoansByCategory/QuickFilters/CheckboxList.vue:79]: Good! Agree, I added a new ticket for this https://kiva.atlassian.net/browse/CORE-820

Comment (dyersituations) [reply src/components/LoansByCategory/QuickFilters/CheckboxList.vue:79]: Thanks!

**Review by dyersituations — APPROVED** [review]

**Pull request merged**